src/test/java/com/fasterxml/jackson/databind/deser/jdk/EnumMapDeserializer5165Test.java

```diff
@@ -0,0 +1,63 @@
+package com.fasterxml.jackson.databind.deser.jdk;
+
+import java.util.EnumMap;
+
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.annotation.JsonSetter;
+import com.fasterxml.jackson.annotation.Nulls;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.exc.InvalidNullException;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import org.opentest4j.AssertionFailedError;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+// For [databind#5165]
+public class EnumMapDeserializer5165Test
+{
+    public enum MyEnum {
+        FOO
+    }
+
+    static class Dst {
+        private EnumMap<MyEnum, Integer> map;
+
+        public EnumMap<MyEnum, Integer> getMap() {
+            return map;
+        }
+
+        public void setMap(EnumMap<MyEnum, Integer> map) {
+            this.map = map;
+        }
+    }
+
+    @Test
+    public void nullsFailTest() {
+        ObjectMapper mapper = JsonMapper.builder()
+                .defaultSetterInfo(JsonSetter.Value.forContentNulls(Nulls.FAIL))
+                .build();
+
+        assertThrows(
```

Comment:
Instead of this, we already have `@JacksonTestFailureExpected` annotation, so let's use that.

And tests should be added under `tofix` package.

Finally, although I like modularity in many ways, in this case it'd be good to combine some if not all tests, to shared mapper configuration (esp. since merging to 3.0 requires some work).